c/cert/test/rules/CON30-C/test.c

```diff
@@ -0,0 +1,98 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <threads.h>
+
+static tss_t k;
+
+void do_free(void *d) { free(d); }
+
+void maybe_free(void *d) {}
+
+void m1() {
+  tss_create(&k, free); // COMPLIANT
+  tss_delete(k);
```

Comment:
I think I need an explanation, why is this compliant? [tss_delete](https://en.cppreference.com/w/c/thread/tss_delete) does not call the registered destructor first?

Reply:
After reading this comment I reworked the way this query worked. Please have another look and let me know if you need to discuss; however I have changed the way the query works to I think be much more in line with this experience in the API.

Reply:
I think the addition of the thread wait is good,

I think maybe we can put a summary of this inline doc:
```
// all calls to `tss_create` must be bookended by calls to tss_delete
  // even if a thread is not created.
```
into the implementation notes just to be extra clear on what the accepted use pattern is, now that its a bit more strict, but other than that, LGTM!

Reply:
Done!